### ISSUE TITLE
Confusing on using the margin parameter m in AdaCos loss

### ISSUE THREAD

**Issue by rose-jinyang** [4uiiurz1/pytorch-adacos#13]
Hello
How are you?
Thanks for contributing this project.
But I am confused on using the margin parameter m in AdaCos loss.
u do not use the parameter m in AdaCos layer.
How can I understand this?
Thanks

**Comment by rose-jinyang** [comment]
Hi
Moreover, I found that your AdaCos layer implementation is different with the following.
https://cpp-learning.com/adacos/#AdaCos
![image](https://user-images.githubusercontent.com/47862419/79887856-21e15900-842e-11ea-886d-a808e8a5d112.png)
How can I understand this?
Please let me know asap.
Thanks

**Comment by rose-jinyang** [comment]
Hi
Could u implement Keras version of AdaCos?
I found that DaisukeIshibe/Keras-Adacos has some issues in dynamic tuning of scale parameter.
Especially, I'm interested in multi-gpu implementation.
Thanks

**Comment by rose-jinyang** [comment]
Hi
Did u train AdaCos model on muli-gpu?
I met an issue when train AdaCos keras version for your torch version.
I think that what multi-gpu change a scale parameter in common may be a reason.
how do u think this?
Thanks

**Comment by ReverseSystem001** [comment]
> Hello
> How are you?
> Thanks for contributing this project.
> But I am confused on using the margin parameter m in AdaCos loss.
> u do not use the parameter m in AdaCos layer.
> How can I understand this?
> Thanks
The paper 3.3 says that we will focus on automatically tuning the scale parameter s in the reminder ofthis paper. It seems like that the author does not use margin m

**Comment by BRO-HAMMER** [comment]
Edit: I found the explanation:
"We therefore propose to automatically tune the scale parameter s and **eliminate the margin parameter m from our loss function**, which makes our proposed AdaCos loss different from state-of-the-art softmax loss variants with angular margin."
